doc/specs/#3327 - Application Theming/#3327 - Application Theming.md

```diff
@@ -0,0 +1,500 @@
+
+Additionally, the user should be able to easily switch from one installed theme
+to another. The user should be able to copy a simple blob of settings from the
+web and paste it into their settings to be able to easily try the theme out.
+
+### Difference between "Themes" and "Schemes"
+
+The avid follower of the Windows Terminal might know that the Terminal already
+contains support for "color schemes". What makes themes different from these
+schemes, and why should they be separate objects?
+
+**Color Schemes** are objects that generally control the appearance of the
+Terminal Control itself (the proverbial "black rectangle with text in it").
+Primarily, color schemes are used for setting the "color table" of a terminal
+instance, setting the values for each of the 16 colors in the terminal's color
+table, and the default foreground and background colors. These are properties
+that only apply to the contents of the terminal itself, and not necessarily the
+entire application. Individual terminal control instances can have different
+color schemes. Furthermore, these schemes are largely in-line with schemes
+available on other platform's terminals. These schemes were heavily inspired by
+the great work done at [iTerm2-Color-Schemes].
+
+Alternatively, **Themes** are sets of properties that apply primarily to the
+window of the application itself, but not necessarily the terminal content.
+These properties apply globally to the entire window, as opposed to controlling
+the appearance of individual terminals. These properties include things such as
+the coloration and styling of the tabs in the tab row.
+
+### Theme objects
+
+Themes will be implemented largely similar to the color schemes implementation.
+Currently, the terminal contains a list of available color schemes, and profiles
+can chose to apply a scheme from the list of schemes. We'll add a list of
+`themes`, and globally, the user will be able to specify one of these themes to
+apply.
+
+Take for example the following settings excerpt:
+
+```json
+{
+    "theme": "My Boxy Theme",
+    "themes": [
+        {
+            "name": "My Boxy Theme",
+            "window.applicationTheme": "dark",
+            "tab.radius": 0,
+            "tab.padding": 5,
+            "tab.background": "terminalBackground",
+            "tab.textColor": "key:SystemAccentColorLight3",
+            "tab.icon": "outline",
+            "tab.closeButton": "hidden",
+            "tabRow.background": "accent",
+            "tabRow.shadows": false
+        },
+        {
+            "name": "My small light theme",
+            "tabBackground": "#80ff0000",
+            "tabRowBackground": "#ffffffff",
+            "tabHeight": 8,
+            "applicationTheme": "light",
+            "colorScheme": "Solarized Light",
+            "tabIcon": "hidden",
+            "tabCloseButton": "hover"
+        }
+    ]
+}
+```
+
+> **TODO FOR DISCUSSION**: I've given both a `tab.<property>` and a

```

Comment:
I personally like the `<element>.<property>` representation more. I think it's more clear.

I'm probably in the minority here but I also _love_ nested objects too. So, is there any consideration for...
```
<element>: 
{
  <property1>: <value1>,
  <property2>: <value2>
}
```

Reply:
ew, I hate the nested objects so much that I didn't even put them in here 😝 My preference is definitely leaning towards `element.property`